api.md

```diff
@@ -0,0 +1,188 @@
+API Documentation
+===
+* [**Builder**](#builder)
+* [**Expression Strings**](#module-tree-expressions)
+
+Builder
+---
+* [**new Builder()**](#new-builderconfig)
+* [**config()**](#builderconfigconfig-saveforreset-ignorebaseurl)
+* [**loadConfig()**](#builderloadconfigconfigfile-saveforreset-ignorebaseurl)
+* [**loadConfigSync()**](#builderloadconfigsyncconfigfile-saveforreset-ignorebaseurl)
+* [**reset()**](#builderconfigconfig-saveforreset-ignorebaseurl)
+* [**bundle()**](#builderconfigconfig-saveforreset-ignorebaseurl)
+* [**buildStatic()**](#builderconfigconfig-saveforreset-ignorebaseurl)
+* [**trace()**](#builderconfigconfig-saveforreset-ignorebaseurl)
+* [**addTrees()**](#builderaddtreesfirsttree-secondtree)
+* [**subtractTrees()**](#buildersubtracttreesfirsttree-secondtree)
+* [**intersectTrees()**](#builderintersecttreesfirsttree-secondtree)
+
+### new Builder([config])
+### new Builder([baseURL, [configFile]])
+`config`: An object conforming to the [config api] (https:/github.com/systemjs/systemjs/blob/master/docs/config-api.md)  
+`baseURL`: Sets the root for the loader  
+`configFile`: A systemjs config file conforming to the [systemjs config api] (https:/github.com/systemjs/systemjs/blob/master/docs/config-api.md)  
+#### Example
+```javascript
+new Builder({
+  map: {
+    'a': 'b.js'
+  }
+});
+
+new Builder('scripts', 'config.js');
+```
+
+### builder.config(config[, saveForReset[, ignoreBaseURL]])
+`config`: An object conforming to the [config api] (https:/github.com/systemjs/systemjs/blob/master/docs/config-api.md)  
+`saveForReset`: Reload this config when builder.reset() is called  
+`ignoreBaseURL`: Don't use the baseURL property from this config  
+#### Example
+```javascript
+builder.config({
+  map: {
+    'a': 'b.js'
+  }
+});
+```
+
+### builder.loadConfig(configFile[, saveForReset[, ignoreBaseURL]])
+`configFile`: A systemjs config file conforming to the [systemjs config api] (https:/github.com/systemjs/systemjs/blob/master/docs/config-api.md)  
+`saveForReset`: Reload this config when builder.reset() is called  
+`ignoreBaseURL`: Don't use the baseURL property from this config  
+#### Example
+```javascript
+builder.loadConfig('config.js').then(() => {
+});
+```
+### builder.loadConfigSync(configFile[, saveForReset[, ignoreBaseURL]])
+Synchronous version of `builder.loadConfig()`  
+
+`configFile`: A systemjs config file conforming to the [systemjs config api] (https:/github.com/systemjs/systemjs/blob/master/docs/config-api.md)  
+`saveForReset`: Reload this config when builder.reset() is called  
+`ignoreBaseURL`: Don't use the baseURL property from this config  
+#### Example
+```javascript
+builder.loadConfigSync('config.js');
+```
+
+### builder.reset()
+Reset the builder config to its initial state, or the last saved config() state
```

Comment:
It's important to mention here that this clears all the builder loader registries by creating new instances, but maintains the builder cache between instances.